### PR TITLE
doc/blog: announce crdb

### PR DESCRIPTION
**Pull request by hedwigz**

### CODE REVIEW

File: doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  
+CockroachDB has been gaining popularity and many of you [have](https://github.com/ent/ent/issues/2545) [been](https://github.com/ariga/atlas/issues/785#issue-1231951038) [asking](https://github.com/ariga/atlas/issues/785#issuecomment-1125853135) us about it.
+
+While CockroachDB aims to be PostgreSQL compatible, it still has some incompatibilities (e.g. [1](https://github.com/cockroachdb/cockroach/issues/20296#issuecomment-1066140651),[2](https://github.com/cockroachdb/cockroach/issues/82064),[3](https://github.com/cockroachdb/cockroach/issues/81659)) which prevent the naive postgres dialect from working with it. Due to this, we had to work around these issues.  
+  
+With the latest release of Atlas, the postgres dialect automatically detects when it is connected to CRDB and uses a custom Driver to handle that.
+
+# How to use
+## Atlas CLI
+Make sure you are using the latest version of Atlas, then connect to your CRDB as if it was Postgres, for example:
```
Comment (hilakashai) [inline doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md:25]:
```suggestion
Make sure you are using the latest version of Atlas, then connect to your CockroachDB as if it were a Postgres database, for example:
```

File: doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+```bash
+# to save the state of 'public' schema to `schema.hcl` run:
+atlas schema inspect -u "postgres://root:pass@localhost:26257/defaultdb?sslmode=disable" -s "public" > schema.hcl
+```
+## Ent
+If you use Ent, make sure you use the latest version, then connect to your CRDB as if it was Postgres, for example:
```
Comment (hilakashai) [inline doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md:31]:
```suggestion
If you use Ent, make sure you use the latest version, then connect to your CockroachDB as if it were a Postgres database, for example:
```

File: doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+if err := client.Schema.Create(
+  context.Background(),
+  schema.WithAtlas(true),
+); err != nil {
+  log.Fatal("opening ent client", err)
+}
+```
+
+# Preview support
+The integration of Atlas with CRDB is well tested with CRDB version `v21.2.11` (at the time of writing, `latest`) and we will extend that in the future.
```
Comment (hilakashai) [inline doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md:46]:
```suggestion
The integration of Atlas with CockroachDB is well tested with version `v21.2.11` (at the time of writing, `latest`) and will be extended in the future.
```

File: doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+If you're using other versions of CRDB or looking for help, don't hesitate to [file an issue](https://github.com/ariga/atlas/issues) or join our [Discord channel](https://discord.gg/zZ6sWVg6NT).
```
Comment (hilakashai) [inline doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md:47]:
```suggestion
If you're using other versions of CockroachDB or looking for help, don't hesitate to [file an issue](https://github.com/ariga/atlas/issues) or join our [Discord channel](https://discord.gg/zZ6sWVg6NT).
```

**Review by hilakashai — COMMENTED** [review]
Looks good but I'd add a little bit more to the overall story

File: doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+personally worked on. As of v0.4.2, Atlas includes preview support for CockroachDB 🎉 
+
+## Atlas
+[Atlas](https://atlasgo.io) is an open-source project that helps developers better manage their database
+schemas. It has a [CLI tool](https://atlasgo.io/cli/reference) and a
+[Terraform integration](https://atlasgo.io/blog/2022/05/04/announcing-terraform-provider). By using Atlas's
+Data Definition Language (with a syntax similar to Terraform), users can plan, verify and apply changes
+to their databases in a simple, declarative workflow.
+Earlier this year, Atlas became the [migration engine for Ent](https://entgo.io/blog/2022/01/20/announcing-new-migration-engine),
+a widely popular, Linux Foundation backed entity framework for Go.
```
Comment (masseelch) [inline doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md:19]:
```suggestion
a popular, Linux Foundation backed entity framework for Go.
```

Not sure if there is something like "widley" popular. Either it is popular id it is not.

File: doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+CockroachDB has been gaining popularity and many of you [have](https://github.com/ent/ent/issues/2545)
+[been](https://github.com/ariga/atlas/issues/785#issue-1231951038) [asking](https://github.com/ariga/atlas/issues/785#issuecomment-1125853135) 
+for Atlas to support it.
+
+While CockroachDB aims to be PostgreSQL compatible, it still has some incompatibilities 
+(e.g. [1](https://github.com/cockroachdb/cockroach/issues/20296#issuecomment-1066140651), 
+[2](https://github.com/cockroachdb/cockroach/issues/82064),[3](https://github.com/cockroachdb/cockroach/issues/81659))
+which prevented Atlas users from using the existing Postgres dialect from working with it.  
+  
+With the latest release of Atlas, the Postgres driver automatically detects when it is connected to a CockroachDB
```
Comment (masseelch) [inline doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md:38]:
```suggestion
With the latest release of Atlas, the Postgres driver automatically detects if it is connected to a CockroachDB
```

File: doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+Next, use Atlas's `schema inspect` command to read the schema of our local database and save the result to a file:
+```
+atlas schema inspect -u 'postgres://root:pass@localhost:26257/?sslmode=disable' --schema public > schema.hcl
+```
+Observe the current HCL representation of the `public` schema, empty with no tables:
+```hcl
+schema "public" {
+}
+```
+
```
Comment (masseelch) [inline doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md:65]: This seems like an unnecessary step. Once can just create the file manual in the step below, where it gets edited anyway. Inspect afterwards is cooler I think. It demonstrates more of what the tool can do.

File: doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+```
+-- Planned Changes:
+-- Create "test" table
+CREATE TABLE "public"."test" ("id" integer NOT NULL, PRIMARY KEY ("id"))
+Use the arrow keys to navigate: ↓ ↑ → ←
+? Are you sure?:
+  ▸ Apply
+    Abort
+```
+After hitting "Apply", Atlas connects applies our desired schema to the database:
```
Comment (masseelch) [inline doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md:95]:
```suggestion
After hitting "Apply", Atlas applies the desired schema to the database:
```

**Review by masseelch — APPROVED** [review]

File: doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
+---
+title: Announcing Atlas CLI v0.4.2 with preview support for CockroachDB 🪳
```
[the remark this answers is not in the document]
Comment (hedwigz) [reply doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md:2]: cockroach emoji, works on the website

File: doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+Start by downloading the latest version of Atlas, on macOS:
+```
+brew install ariga/tap/atlas
+```
+For installation instructions on other platforms, see [the docs](https://atlasgo.io/cli/getting-started/setting-up#install-the-cli).
+
+For the purpose of this example, let's spin up a local, [single-node CockroachDB cluster](https://github.com/cockroachlabs-field/cockroachdb-single-node)
+in a container by running:
+```
+docker run -it -p 8080:8080 -p 26257:26257 -e "DATABASE_NAME=test" -e "MEMORY_SIZE=.5" timveil/cockroachdb-single-node:latest
```
Comment (hedwigz) [inline doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md:53]:
```suggestion
docker run -it -p 8080:8080 -p 26257:26257 -e "DATABASE_NAME=test" -e "MEMORY_SIZE=.5" cockroachdb/cockroach:v21.2.11 start-single-node --insecure
```

File: doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+```
+docker run --rm -d -p 26257:26257 --name crdb cockroachdb/cockroach start-single-node --insecure
+```
+
+Next, let's seed the database with a simple table:
+```
+docker exec crdb cockroach sql --insecure -e 'CREATE TABLE users (id int primary key);'
+```
+
+Next, use Atlas's `schema inspect` command to read the schema of our local database and save the result to a file:
```
Comment (masseelch) [inline doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md:61]: Next, ... Next

File: doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+> ACID transactions; and provides a familiar SQL API for structuring, manipulating, and querying data.  
+  
+CockroachDB has been gaining popularity and many of you [have](https://github.com/ent/ent/issues/2545)
+[been](https://github.com/ariga/atlas/issues/785#issue-1231951038) [asking](https://github.com/ariga/atlas/issues/785#issuecomment-1125853135) 
+for Atlas to support it.
+
+While CockroachDB aims to be PostgreSQL compatible, it still has some incompatibilities 
+(e.g. [1](https://github.com/cockroachdb/cockroach/issues/20296#issuecomment-1066140651), 
+[2](https://github.com/cockroachdb/cockroach/issues/82064),[3](https://github.com/cockroachdb/cockroach/issues/81659))
+which prevented Atlas users from using the existing Postgres dialect from working with it.
```
[the remark this answers is not in the document]
Comment (rotemtam) [reply doc/website/blog/2022-06-announcing-cockroachdb-support-preview.md:36]: thx <3

File: doc/website/blog/2022-06-09-announcing-cockroachdb-support-preview.md [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+
+For the purpose of this example, let's spin up a local, [single-node CockroachDB cluster](https://www.cockroachlabs.com/docs/stable/cockroach-start-single-node.html)
+in a container by running:
+```
+docker run --rm -d -p 26257:26257 --name crdb cockroachdb/cockroach start-single-node --insecure
+```
+
+Next, let's seed the database with a simple table:
+```
+docker exec crdb cockroach sql --insecure -e 'CREATE TABLE users (id int primary key);'
```
Comment (masseelch) [inline doc/website/blog/2022-06-09-announcing-cockroachdb-support-preview.md:58]: :+1:

**Review by masseelch — APPROVED** [review]

**Pull request merged**